editor/src/clj/editor/app_manifest.clj

```diff
@@ -0,0 +1,478 @@
+  {:setting :check-box :toggles toggles})
+
+(defn make-choice-setting [& kw+toggles-then-none]
+  (let [none (last kw+toggles-then-none)
+        choices (into [] (partition-all 2) (butlast kw+toggles-then-none))]
+    (assert (keyword? none))
+    (assert (every? (fn [[kw toggles]]
+                      (and (keyword? kw) (coll? toggles) (every? map? toggles)))
+                    choices))
+    {:setting :choice
+     :choices choices
+     :none none}))
+
+(defn get-setting-value [manifest setting]
+  (case (:setting setting)
+    :check-box
+    (->> setting
+         :toggles
+         (mapv #(get-toggle-value manifest %))
+         properties/unify-values)
+
+    :choice
+    (let [{:keys [choices none]} setting
+          first-fit-choice (some (fn [[_kw toggles :as choice]]
+                                   (when (properties/unify-values (mapv #(get-toggle-value manifest %) toggles))
+                                     choice))
+                                 choices)]
+      (if first-fit-choice
+        (let [[choice-kw choice-toggles] first-fit-choice
+              choice-toggle? (set choice-toggles)
+              remaining-toggle-values (into []
+                                            (comp
+                                              (mapcat second)
+                                              (remove choice-toggle?)
+                                              (map #(get-toggle-value manifest %)))
+                                            choices)]
+          (if (or (zero? (count remaining-toggle-values))
+                  (false? (properties/unify-values remaining-toggle-values)))
+            choice-kw
+            nil))
+        (let [all-toggles-unify-to-nil (nil? (properties/unify-values
+                                               (into []
+                                                     (comp
+                                                       (mapcat second)
+                                                       (map #(get-toggle-value manifest %)))
+                                                     choices)))]
+         (if all-toggles-unify-to-nil nil none))))))
+
+(defn set-setting-value [manifest setting value]
+  (case (:setting setting)
+    :check-box (reduce #(set-toggle-value %1 %2 value) manifest (:toggles setting))
+    :choice (let [{:keys [choices none]} setting
+                  enabled-toggles (if (identical? none value)
+                                    nil
+                                    (some (fn [[kw toggles]]
+                                            (when (identical? kw value) toggles))
+                                          choices))
+                  disabled-toggles (if (identical? none value)
+                                     (mapcat second choices)
+                                     (into []
+                                           (comp
+                                             (remove #(identical? value (first %)))
```

Comment:
Makes sense. My reasoning was that all checkbox settings are defined in code, so values are probably keywords, but we don't really lose anything by doing `=` instead of `identical?` there, it shouldn't be a part of a hot loop. I'll fix it!